### DISCUSSION TITLE
Gateway Port Mapping Strategy

### DISCUSSION THREAD

**Discussion by benmusson** [mussonindustrial/testcontainers-ignition#19] [Ideas]
## Background
Right now (0.1.0), several well known ports are made available:

1. `8088` (Gateway HTTP)
2. `8043` (Gateway HTTPS)
3. `8060` (GAN)
4. `8000` (Debug)

The IgnitionContainer builder doesn't allow for changing these ports when starting the Gateway. The ports are kept at their default values, and several methods are defined to return the mapped testcontainers alias (`getGatewayHttpPort`, `getGatewayHttpsPort`, `getGatewayGanPort`, and `getDebugPort`). 

## Questions
1. Should we included the ability to reassign the well-known ports when starting the Gateway?
I.e. assign the gateway's HTTP port to `9088` when creating the docker container. Is this useful in a testing context?

2. Do the convenience methods (`getGatewayHttpPort`, `getGatewayHttpsPort`, `getGatewayGanPort`, and `getDebugPort`) need to exist? They are named incorrectly; they return the testcontainers _mapped_ port, not the raw port used by the Gateway, but `getMappedGatewayHttpPort` seems kinda goofy and not any more helpful than simply using `getMappedPort(8088)` or `getMappedPort(GatewayHttpPort)`.
 
3. What other well known ports are worth adding to this library? Definitely `62541` (OPC UA Server), what else?

**Comment by benmusson** [comment]
@kevinherron pinging for opinion

**Reply by kevinherron** [reply]
I was originally going to add 62541 to the default exposed ports but decided against it because the server is configured to bind only to localhost by default.

For me, the benefit of the methods (name aside) is to aid in discovery - I initially missed that you could just manually add exposed ports via super class. Not a big deal either way now that I know about it though :shrug:
